enrysrv/cli/common.go

```diff
@@ -0,0 +1,57 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+
+	"github.com/sirupsen/logrus"
+	"gopkg.in/src-d/go-errors.v1"
+)
+
+var (
+	ErrMaxMessageSizeTooBig = errors.NewKind("max-message-size too big (limit is 2047MB): %u")
+)
+
+type commonCmd struct {
+	LogLevel       string `long:"log-level" description:"log level" default:"debug"`
+	MaxMessageSize string `long:"max-message-size" description:"maximum message size to send/receive to/from clients (in MB)" default:"100"`
+	Version        bool   `short:"v" long:"version" description:"print version"`
+}
+
+var versionMessage = fmt.Sprintf("version:%s build:%s commit:%s", version, build, commit)
+
+func (c *commonCmd) exec(args []string) error {
+	if c.Version {
+		fmt.Fprintf(os.Stdout, "%s\n", versionMessage)
+		return nil
+	}
+
+	level, err := logrus.ParseLevel(c.LogLevel)
+	if err != nil {
+		return err
+	}
+
+	logrus.SetLevel(level)
+	logrus.Info(versionMessage)
+	return nil
+}
+
+func (c *commonCmd) parseMaxMessageSize() (int, error) {
+	if c.MaxMessageSize == "" {
+		return 0, nil
+	}
+
+	n, err := strconv.ParseUint(c.MaxMessageSize, 10, 16)
+	if err != nil {
+		return 0, err
+	}
+
+	if n >= 2048 {
+		// Setting the hard limit of message size to less than 2GB since
```

Comment:
This code looks tremendously similar to https://github.com/bblfsh/server/blob/cf9cc98d152c4e4918a44f31862597fe338e00ab/cmd/bblfsh/common.go#L39

Babelfish server is a project under GPL-3 https://github.com/bblfsh/server/blob/master/LICENSE and Berserker does not have a license yet #12 so, it's proprietary code now.

You seems not to be the copyright holder (original author) of these lines in Babelfhish, so the only way to **legally** copy-paste somebody's else GPL code - is to paste it to GPL project, as it's copyleft.

So, to merge this PR, one for the 2 things must happen:
 - either with @mcuadros  help we decide Berserker to become GPL3 in #12 
 - or you need to re-write  this code, without copying from GPL3 project

Just for the future, to simplify such cases - in case you copy a code from somewhere, it's always better to say so in comments, just in case the license of the original code requires to keep attribution of original authors